www/activities/diary/js/diary.js

```diff
@@ -220,6 +245,9 @@ app.Diary = {
     if (requestedScrollPosition !== undefined) {
       $(".page[data-name='diary'] .page-content").scrollTop(requestedScrollPosition);
     }
+
+    // Send statistics. This is the wrong place. but idk. a better place.
+    this.sendStatistics();
```

Comment:
This should probably be an action in the import/export page

Reply:
"sendStatistics" will send the data to the external service.

With moving it into import/export page it will only sent it to the external service on loading the import / export page.

Reply:
Placing the function here will execute it every time a diary item is added, removed, or edited, but also when you just open the diary page. So this is probably not the behaviour you want.
To fix this, see where the render function is called from and consider if adding your function call there makes sense.

Reply:
I would suggest to place the "sendStatistics" function calls in these places:
- https://github.com/Kroseida/waistline/blob/6df9f359273325f6e239a0a6e7df0241d95f7e69/www/activities/diary/js/diary.js#L517
- https://github.com/Kroseida/waistline/blob/6df9f359273325f6e239a0a6e7df0241d95f7e69/www/activities/diary/js/diary.js#L618
- https://github.com/Kroseida/waistline/blob/6df9f359273325f6e239a0a6e7df0241d95f7e69/www/activities/diary/js/diary.js#L641
- https://github.com/Kroseida/waistline/blob/6df9f359273325f6e239a0a6e7df0241d95f7e69/www/activities/diary/js/diary.js#L677
- https://github.com/Kroseida/waistline/blob/6df9f359273325f6e239a0a6e7df0241d95f7e69/www/activities/diary/js/diary.js#L765

Reply:
Exactly this was the purpose I want.
My problem is just the point: sending data in a render function is a little bit weird.

But user should be able to send data to their service by just opening the page, so they can just "sync" old data.
The Backend Part / Service will handle everything else and make Duplications to updates.
I can also change as you suggested. But I would recommend to send the data on every action. (Also loading the page.)
So you can also sync the data if you have no internet connection while adding entries.

Pro:
- Better Data Sync (also for old data)

Con:
- More Requests

Offtopic: I am currently working on the service part as an open-source solution. I could push it to my personal github page or if you have an group I could also push it in your group.

Reply:
That reminds me, it would probably be a good idea to check whether the device is connected to the internet before calling the "sendStatistics" (or checking it inside the function). And maybe it should even only send the data when connected to Wi-Fi to save mobile bandwidth.

Reply:
maybe this should be a setting entry too?

Reply:
I think for the foreseeable future, you will be the only person using this feature. So decide for yourself whether you need a setting or not :D
But since the app will send data to the service every time you open the diary, only sending when connected to Wi-Fi should be the default, in my opinion. Otherwise this would use up too much mobile data.

Reply:
definitely a good point! (This feature is for me and some friends. So we have some more ppl. :P)
I will implement this point as a setting entry. And make it to default true.

Reply:
> Maybe we should move this entire section from "statistics" to "developer features"?

Yes, that sounds like a good idea to me.

Reply:
done: (example)
![image](https://user-images.githubusercontent.com/18089806/189498125-c5bbea3d-4dbb-4cbe-9ad3-5fd12c4df2a5.png)

defaults are:
        "data-sharing-active": false,
        "data-sharing-wifi-only": true,
        "data-sharing-address": "",
        "data-sharing-authorization": ""

@EmilJunker

Reply:
> Wifi could not be tested. I have to learn how to build it for android

I didn't test it either, but the source code looks good to me, so it should work :)